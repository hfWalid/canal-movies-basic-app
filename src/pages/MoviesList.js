import React from 'react';

import { MoviesList } from '../features/moviesList';


const MoviesListPage: function = (props) => {
    return (
        <MoviesList {...props}/>
    );
};

export default MoviesListPage;
