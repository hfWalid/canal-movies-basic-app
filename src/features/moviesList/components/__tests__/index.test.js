import React from 'react';
import { cleanup, render } from 'react-testing-library';
import MoviesList from "../index";
import {Provider} from "react-redux";
import configureStore from "../../../../store";

jest.mock('react-router-dom', () => ({
    useLocation: () => ({
        push: jest.fn(),
    }),
}));

// automatically unmount and cleanup DOM after the test is finished.
afterEach(cleanup);

describe('renders movies list feature', () => {
    let store;
    beforeAll(() => {
        store = configureStore({});
    });

    test('renders movies list feature', async () => {
        const { asFragment } = render(
            <Provider store={store} >
                <MoviesList />
            </Provider>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
