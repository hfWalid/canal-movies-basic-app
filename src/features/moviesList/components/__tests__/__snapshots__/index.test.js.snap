// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders movies list feature renders movies list feature 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL QzUTY"
  >
    <div
      class="sc-gsDKAQ JPWeZ"
    />
    <div
      class="sc-dkPtRN fbvjfo"
    />
  </div>
</DocumentFragment>
`;
